lib/models/data_controller.dart

```diff
@@ -5,19 +5,19 @@ import 'login_model.dart';
 import 'product_model.dart';
 
 final getDataFuture =
-    ChangeNotifierProvider<GetDataFromApi>((ref) => GetDataFromApi());
+    ChangeNotifierProvider<DataController>((ref) => DataController());
```

Comment:
use the provider variable with name 'provider' and use it as static and move inside the class